From 2eeadd3c4f829715a38b166a1efe82b8b0e1bf9e Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Tue, 16 Sep 2014 17:06:33 -0400
Subject: [usb] uas: Fix bounds check in uas_find_endpoints

Message-id: <1410887463-20674-322-git-send-email-dzickus@redhat.com>
Patchwork-id: 95288
O-Subject: [RHEL7 PATCH 321/591] uas: Fix bounds check in uas_find_endpoints
Bugzilla: 1110939
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1110939

commit d495c1baa1b3ba277bb5ae24adeab0600151cba4
Author: Hans de Goede <hdegoede@redhat.com>
Date:   Tue Oct 29 09:06:54 2013 +0100

    uas: Fix bounds check in uas_find_endpoints

    The loop uses up to 3 bytes of the endpoint extra data.

    Signed-off-by: Hans de Goede <hdegoede@redhat.com>
    Signed-off-by: Sarah Sharp <sarah.a.sharp@linux.intel.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/usb/storage/uas.c b/drivers/usb/storage/uas.c
index 1ac66f2..7662b3e 100644
--- a/drivers/usb/storage/uas.c
+++ b/drivers/usb/storage/uas.c
@@ -907,7 +907,7 @@ static int uas_find_endpoints(struct usb_host_interface *alt,
  for (i = 0; i < n_endpoints; i++) {
   unsigned char *extra = endpoint[i].extra;
   int len = endpoint[i].extralen;
-  while (len > 1) {
+  while (len >= 3) {
    if (extra[1] == USB_DT_PIPE_USAGE) {
     unsigned pipe_id = extra[2];
     if (pipe_id > 0 && pipe_id < 5)
-- 
1.7.1